### PR TITLE
fix(shell): double quote params

### PR DESCRIPTION
**Pull request by dargmuesli**
I've set `test/default.sh`'s shebang to `bash` because `echo -e` is used in this file.

### CODE REVIEW

**Comment by typicode** [comment]
Thanks for the PR.
What would be the use case for symlinked hooks?

**Comment by dargmuesli** [comment]
Any application that parses the git hooks directory itself without respecting a custom directory set through git config, like GitKraken for example. I suggested respecting the config setting to them earlier today, but until then the many users of GitKraken won't see husky v5 hooks executed as long as they don't add symlinks to the default git hook directory.

**Comment by dargmuesli** [comment]
Why is the CI failing btw? :0

**Comment by dargmuesli** [comment]
I resolved the merge conflicts. @typicode what's your status on this PR? :)

**Comment by typicode** [comment]
Hmm, I'd prefer to keep the script as it is and not rely on `readlink`. I'd rather not have fixes for external tools in husky. 

However, if users want to edit their `pre-commit` hook to handle specifically some tools, they can edit this line manually.

As for the rest LGTM, good catch on the `"$0"` :+1:

File: .husky/commit-msg [hunk 1]
```diff
@@ -1,4 +1,6 @@
 #!/bin/sh
```
Comment (typicode) [inline .husky/commit-msg:1]: Could your revert this file and `.husky/pre-commit`

File: scripts/pre-commit [hunk 2]
```diff
@@ -4,7 +4,7 @@ echo "husky: installing Git hooks"
 
 mkdir -p "$husky_dir/_"
 echo "_" > "$husky_dir/.gitignore"
-cp "$(dirname $0)/husky.sh" "$husky_dir/_"
+cp "$(dirname "$0")/husky.sh" "$husky_dir/_"
```
Comment (typicode) [inline scripts/pre-commit:7]: :+1:

File: src/commands/add.ts [hunk 3]
```diff
@@ -15,7 +15,12 @@ function createHookFile(file: string, cmd: string) {
     throw new Error(`${file} already exists`)
   }
 
-  const data = ['#!/bin/sh', '. "$(dirname $0)/_/husky.sh"', '', cmd].join('\n')
+  const data = [
```
Comment (typicode) [inline src/commands/add.ts:18]:
```suggestion
    const data = ['#!/bin/sh', '. "$(dirname "$0")/_/husky.sh"', '', cmd].join('\n')
```

File: test/sub-dir.sh [hunk 4]
```diff
@@ -1,4 +1,4 @@
-. $(dirname $0)/_functions.sh
+. $(dirname "$0")/_functions.sh
```
Comment (typicode) [inline test/sub-dir.sh:1]: :+1: could you change `test/config-dir.sh` too?

File: test/default.sh [hunk 5]
```diff
@@ -1,4 +1,7 @@
-. $(dirname $0)/_functions.sh
+#!/bin/bash
```
Comment (typicode) [inline test/default.sh:1]: I think the shebang can be removed since these scripts are not executable.

File: test/default.sh [hunk 6]
```diff
@@ -1,4 +1,7 @@
...
+
+# shellcheck source=./_functions.sh
```
Comment (typicode) [inline test/default.sh:3]: Is this necessary?

File: .husky/commit-msg (hunk 1 above)
Comment (dargmuesli) [reply .husky/commit-msg:1]: I've removed the `readlink` now, or do you want me to revert everything? If yes, why?

File: test/sub-dir.sh (hunk 4 above)
Comment (dargmuesli) [reply test/sub-dir.sh:1]: Done.

File: src/commands/add.ts (hunk 3 above)
Comment (dargmuesli) [reply src/commands/add.ts]: Done, except for the additional indentation (?) :)

File: test/default.sh (hunk 5 above)
Comment (dargmuesli) [reply test/default.sh]: Replaced that with the shellcheck comment.

**Comment by dargmuesli** [comment]
> I'd rather not have fixes for external tools in husky.

I don't really see it as a fix for external tools but as broadened functionality for husky even if the "fix" happened to be the reason for this PR.

**Review by typicode — APPROVED** [review]
Thanks! :)

**Pull request merged**